plugin/trino-kafka-event-listener/src/main/java/io/trino/plugin/eventlistener/kafka/KafkaRecordBuilder.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.eventlistener.kafka;
+
+import com.fasterxml.jackson.annotation.JsonFilter;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.ser.FilterProvider;
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import io.trino.plugin.eventlistener.kafka.metadata.MetadataProvider;
+import io.trino.plugin.eventlistener.kafka.model.QueryCompletedEventWrapper;
+import io.trino.plugin.eventlistener.kafka.model.QueryCreatedEventWrapper;
+import io.trino.spi.eventlistener.QueryCompletedEvent;
+import io.trino.spi.eventlistener.QueryCreatedEvent;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.reflect.ReflectData;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.Serializer;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Set;
+
+public class KafkaRecordBuilder
+        implements Serializer<Object>
+{
+    private static final Schema QUERY_COMPLETED_EVENT_SCHEMA =
+            ReflectData.get().getSchema(QueryCompletedEventWrapper.class);
+    private static final Schema QUERY_STARTED_EVENT_SCHEMA =
+            ReflectData.get().getSchema(QueryCreatedEventWrapper.class);
+
+    private final ObjectWriter writer;
+    private final String startedTopic;
+    private final String completedTopic;
+    private final MetadataProvider metadataProvider;
+
+    @JsonFilter("filter properties by name")
```

Comment:
what does it do? is it needed since `SimpleFilterProvider` is created?

Reply:
That's correct